### PR TITLE
FS socketio setup #30

### PR DESCRIPTION
**Pull request by logan858**
### What this PR does (required):
- initial socketio setup, users emit connection messages to the server upon dashboard load and after authenticating the user's credentials
 
### Screenshots / Videos (required):
image

### Any information needed to test this feature (required):
- Socketio logs online user data on the server side when a user connects to dashboard.  When tracking all online users, presumably we would implement an object or data endpoint where we would push this info into (upon login) and out of (upon logout).
- npm i socket.io-client   on the client side
- npm i socket.io      on server side
- npm i cookies       on server side

### Any issues with the current functionality (optional):
n/a

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -27,8 +29,29 @@ const io = socketio(server, {
   },
 });
 
+io.use(async (socket, next) => {
+  if (socket.handshake.headers.cookie) {
+    const authToken = cookie.parse(socket.handshake.headers.cookie).token;
+    if (authToken) {
+      const foundUser = await jwtVerifyUser(authToken, next)
+      if (foundUser) {
+        console.log(foundUser)
```
Comment (ezeilo-su) [inline server/app.js:38]: Remove console log

File: server/app.js [hunk 2]
```diff
@@ -27,8 +29,29 @@ const io = socketio(server, {
...
+    }
+  } else {
+    next(new Error('Auth Error'))
+  }
+})
+
 io.on("connection", (socket) => {
-  console.log("connected");
+  socket.on('emit-msg', (num, str) => {
+    console.log(num, '\n', str)
```
Comment (ezeilo-su) [inline server/app.js:53]: Remove console log

File: client/src/context/useSocketContext.tsx [hunk 3]
```diff
@@ -19,10 +19,15 @@ export const SocketProvider: FunctionComponent = ({ children }): JSX.Element =>
...
         withCredentials: true,
+        extraHeaders: {
+          'custom-header': 'test',
+        },
       }),
-    );
+      );
   }, []);
 
+  // socket?.emit("emit-msg", 10, 'testing...')
```
Comment (ezeilo-su) [inline client/src/context/useSocketContext.tsx:29]: Remove unused line of code

**Review by sundayezeilo — CHANGES_REQUESTED** [review]
Good work on this one. Just address the few comments left below.

**Pull request merged**